src/storeProtectLoadUtils.js

```diff
@@ -0,0 +1,155 @@
+import { isCmsUi } from '@plone/volto/helpers';
+
+const LOCATION_CHANGE = '@@router/LOCATION_CHANGE';
+const PROTECT_START = '@@loadProtector/START';
+const PROTECT_END = '@@loadProtector/END';
+const PROTECT_SKIPPED = '@@loadProtector/SKIPPED';
+const GET_CONTENT_PENDING = 'GET_CONTENT_PENDING';
+const GET_CONTENT_SUCCESS = 'GET_CONTENT_SUCCESS';
+const GET_CONTENT_FAIL = 'GET_CONTENT_FAIL';
+const RESET_CONTENT = 'RESET_CONTENT';
+
+// ---
+// The load protector middlewares have multiple purposes.
+//
+// 1. Maintained a delayed location that can be used to trigger an UI
+// change (such as scroll to top) after all content for the page has been
+// loaded.
+//
+// 2. When moving from a non content route to a content route, issue
+// a data reset before the content load operation. - However do not do
+// such a reset when navigating between two content routes.
+// ---
+
+export const protectLoadStart = ({ dispatch, getState }) => (next) => (
+  action,
+) => {
+  if (typeof action === 'function') {
+    return next(action);
+  }
+  switch (action.type) {
+    case LOCATION_CHANGE:
+      const { location } = action.payload;
+      const { pathname: path } = location;
+      const currentPath = getState().router.location.pathname;
+      if (isCmsUi(path)) {
+        // Next path: isCmsUI, Non Content. There is no
+        // loading here, so skip counting altogether.
+        // Will update the delayed location constantly.
+        dispatch({
+          type: PROTECT_SKIPPED,
+          location,
+        });
+      } else {
+        dispatch({
+          type: PROTECT_START,
+          location,
+          // Only reset before the fetch, if we depart from
+          // a not isCmsUi, Content pass. However, reset will
+          // not occur if moving between two content paths,
+          // only the postponed location will be booked.
+          resetBeforeFetch: isCmsUi(currentPath),
+        });
+      }
+      return next(action);
+    default:
+      return next(action);
+  }
+};
+
+// Note that there is a bit of heuristics here. We assume that every action
+// like this is beginning/ending an action. If this logic fails then the counting
+// will be off, resulting either a premature or an indefinitely missing state change.
+const isRequestAction = (action) => action.type.endsWith('_PENDING');
```

Comment:
Test for `action.request` presence. The api middleware doesn't produce any SUCCES/FAIL/PENDING action for non-request actions. See https://github.com/plone/volto/blob/712e7a6bd207e0da2fa2e54f0952f11d4a78bfdf/src/middleware/api.js#L114-L116

Reply:
Nevermind, I think the use case is different.